src/main/java/com/adobe/aem/compgenerator/utils/DialogUtils.java

```diff
@@ -69,11 +73,56 @@ public static void createDialogXml(final GenerationConfig generationConfig, fina
         }
     }
 
+    /**
+     * Creates the tab dialog properties.
+     *
+     * @param doc The {@link Document} object
+     * @param rootElement the root element
+     * @param tabs The {@link Tab} object
+     */
+    private static void createDialogProperties(Document doc, Element rootElement, List<Tab> tabs,
+            List<Property> properties) {
+
+        if (null != properties && !properties.isEmpty()) {
+            if (null != tabs && !tabs.isEmpty()) {
+                Node currentNode = createTabsParentNodeStructure(doc, rootElement);
+                
+                Map<String, Property> propertiesMap = properties.stream()
+                        .collect(Collectors.toMap(Property::getField, Function.identity()));
+                
+                for (Tab tab : tabs) {
+                    Node tabNode = createTabStructure(doc, tab, currentNode);
+                    List<Property> sortedProperties = tab.getFields().stream().map(propertiesMap::get)
+                            .collect(Collectors.toList());
+                    createNodeStructure(doc, sortedProperties, tabNode);
+                }
+            } else {
+                Node currentNode = updateDefaultNodeStructure(doc, rootElement);
+                createNodeStructure(doc, properties, currentNode);
+            }
+        }
+
+    }
+
+    /**
+     * Creates the node structure.
+     *
+     * @param doc The {@link Document} object
+     * @param properties {@link Property}
+     * @param currentNode the current node
+     */
+    private static void createNodeStructure(Document doc, List<Property> properties, Node currentNode) {
+        properties.stream().filter(Objects::nonNull)
+                .map(property -> createPropertyNode(doc, currentNode, property)).filter(Objects::nonNull)
+                .forEach(a -> currentNode.appendChild(a));
+    }
+
     /**
      * Generates the root elements of what will be the _cq_dialog/.content.xml.
      * 
      * @param document The {@link Document} object
-     * @param generationConfig The {@link GenerationConfig} object with all the populated values
+     * @param generationConfig The {@link GenerationConfig} object with all the
```

Comment:
A handful of lines here and downward that are arbitrarily wrapping too short (per a previous comment) - let's try to reduce these diffs (for lines of code you didnt actually change) so that we dont get all sorts of back and forth diffs from different devs.

Reply:
Sure, Updated my preferences to use 120 going forward.